libraries/lib-wave-track/AudioSegmentSampleView.h

```diff
@@ -0,0 +1,65 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSegmentSampleView.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#pragma once
+
+#include "SampleCount.h"
+
+#include <chrono>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+using BlockSampleView = std::shared_ptr<std::vector<float>>;
+
+//! An audio segment is either a whole clip or the silence between clips.
+class WAVE_TRACK_API AudioSegmentSampleView final
+{
+public:
+   /*
+    * @brief Constructs an `AudioSegmentSampleView` from `BlockSampleView`s.
+    * @param blockViews A continuous sequence of samples, segmented in blocks.
+    * @param start Index of the first sample in `blockViews` to look at.
+    * @param length The number of samples from `start` to look at.
+    * @pre `!blockViews.empty()`
+    * @pre `start < blockViews[0].size()`
+    * @pre `length < sum(sizes(blockViews)) - start`
+    */
+   AudioSegmentSampleView(
+      std::vector<BlockSampleView> blockViews, size_t start,
+      sampleCount length);
+
+   /*
+    * @brief Constructs a silent `AudioSegmentSampleView`.
+    * @param length The number of "silence samples".
+    */
+   AudioSegmentSampleView(sampleCount length);
```

Comment:
This function is not intended to work with long lengths.  The argument should be `size_t`.  Where it is used in `WaveTrack::GetOneSampleView`, there is a use of `as_size_t` which could use some comments.

Reply:
Good that you noticed the `as_size_t` to in the end call the implicit `sampleCount` ctor.
It probably will never be used for more than `size_t` can represent, yet semantically this is an audio segment between two clips, which `sampleCount` can help realize. (Also, I'd have to change it in the other ctor for consistency.)